### PR TITLE
Remove checkstyle suppressions for tests and samples Formrecognizer

### LINKED ISSUE #19177: Enable Spotbugs check on Samples and test code files.

### PR DESCRIPTION
**Pull request by samvaity**
Closes #19177

Build ref - https://dev.azure.com/azure-sdk/public/_build/results?buildId=738801&view=results

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -89,7 +89,7 @@
   <suppress checks="FinalClass" files="Util.java"/>
 
   <!-- Star imports are ok in unit tests - it makes it easier to import all assertions, etc -->
-  <suppress checks="AvoidStarImport" files=".*[/\\]src[/\\]test[/\\]java[/\\].*\.java"/>
+<!--  <suppress checks="AvoidStarImport" files=".*[/\\]src[/\\]test[/\\]java[/\\].*\.java"/>-->
```
Comment (samvaity) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:92]: Will revert these file changes once a green CI proving samples and test files cleaned up.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -22,9 +22,9 @@
   <!--  <suppress checks="com\.azure\.tools\.checkstyle\.checks\..+" files=".*[/\\]com[/\\]microsoft[/\\].*"/>-->
 
   <!-- ignore missing javadoc in samples -->
-  <suppress checks="MissingJavadocMethod" files=".*[/\\]samples[/\\].*\.java"/>
```
Comment (mssfang) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:25]: Do we really want to unable these rules?

Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:25]: yes eventually for all libs. 
Ref - https://github.com/Azure/azure-sdk-for-java/issues/9360

But package owners need to start at their own respective areas and then this suppression can finally be removed.

Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:25]: I will do this for TA and MA too. jfyi

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 3]
```diff
@@ -225,8 +225,9 @@ for (int i = 0; i < recognizedForms.size(); i++) {
 ### Recognize Content
 Recognize text, table structures and selection marks like radio buttons and check boxes, along with their bounding box
 coordinates, from documents, without the need to train a model.
-<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L116-L144 -->
+<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L118-L147 -->
 ```java
+ic void recognizeContent() throws IOException {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:230]: wrong line

**Review by mssfang — APPROVED** [review]

**Pull request merged**